dds/idl/typeobject_generator.cpp

```diff
@@ -225,13 +225,35 @@ typeobject_generator::gen_struct(AST_Structure*, UTL_ScopedName* name,
     const string decl_gto = "getMinimalTypeObject<" + clazz + ">";
     Function gto(decl_gto.c_str(), "const XTypes::TypeObject&", "");
     gto.endArgs();
-
+    const ExtensibilityKind exten = be_global->extensibility(node);
+    string bitwise_str;
+    switch (exten) {
+      case extensibilitykind_final:
+        bitwise_str = "XTypes::IS_FINAL";
+        break;
+      case extensibilitykind_appendable:
+        bitwise_str = "XTypes::IS_APPENDABLE";
+        break;
+      case extensibilitykind_mutable:
+        bitwise_str = "XTypes::IS_MUTABLE";
+        break;
+      default:
+        idl_global->err()->misc_error(
+          "Unexpected extensibility while setting flags", node);
+        return false;
+    }
+    if (be_global->is_nested(node)){
```

Comment:
```suggestion
    if (!be_global->is_topic_type(node)) {
```

This isn't obvious and maybe even a little misleading on my part since I wrote it, but `is_nested` isn't what you want here. I could talk about it more, but the gist is that OpenDDS' relationship with the term "nested "is a little messy. When the spec says "nested", for working in OpenDDS it's better to think of it as "not a topic type". One of the consequences of this is that `is_nested` is referring to an incomplete "nested" property used to determine if a type is a topic type.